src/main/java/abstracts/Car.java

```diff
@@ -0,0 +1,217 @@
+package abstracts;
+
+import interfaces.*;
+import org.apache.log4j.Logger;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Set;
+
+public abstract class Car implements Vehicle {
+
+    /**
+     * log4j var for logging
+     */
+    final static Logger logger = Logger.getLogger(Car.class);
```

Comment:
This Logger is a constant (static final attribute) so its name should be all uppercase:
`static final Logger LOGGER = Logger.getLogger(Car.class);`